flexmeasures/data/queries/annotations.py

```diff
@@ -8,26 +8,33 @@
 from flexmeasures.data.models.annotations import (
     Annotation,
     GenericAssetAnnotationRelationship,
+    SensorAnnotationRelationship,
 )
 from flexmeasures.data.models.data_sources import DataSource
 
 
 def query_asset_annotations(
-    asset_id: int,
+    asset_or_sensor_id: int,
+    relationship_module,
     annotations_after: datetime | None = None,
     annotations_before: datetime | None = None,
     sources: list[DataSource] | None = None,
     annotation_type: str | None = None,
 ) -> Query:
     """Match annotations assigned to the given asset."""
-    query = (
-        select(Annotation)
-        .join(GenericAssetAnnotationRelationship)
-        .filter(
-            GenericAssetAnnotationRelationship.generic_asset_id == asset_id,
+    query = select(Annotation)
+    if relationship_module is GenericAssetAnnotationRelationship:
```

Comment:
Part of the logic in the if/else-block could be factored out.
```suggestion
    query = query.join(relationship_module, relationship_module.annotation_id == Annotation.id)
```